### PR TITLE
Add Trino 397 release notes

### LINKED ISSUE #14154: Release notes for 397

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 397 release. Fixes #14154 

## Non-technical explanation

N/A

## Release notes

(x) This is not user-visible and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

## Verification for each commit

Format: commit, committer/maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 15 September 2022 ✅ 

45eb8ef76630e475eb40dde5a52b7ec635b34865 @arhimondr #14069 ✅ rn ✅ docs
3109aaf2bfdae04fad771be9d3588e6d49434c03 @arhimondr #14069 ✅ rn ✅ docs
9ae9444665dd45ccf5c9d51f77f15b26f2da1f27 @arhimondr #14069 ✅ rn ✅ docs
e1f688d17cbce044c1a5b0a223631a15c8de9185 @arhimondr #14069 ✅ rn ✅ docs

## 16 September 2022 ✅ 

1dc391a6802597d51b83d614e79c8e07e2979a2c @findepi #14145 ✅ rn ✅ docs
315bcb09a928d0a193ff3c21c359de66cdda7106 @findepi #14158 ✅ rn ✅ docs
e9e5e11d0aff1b3117ea6a767c3d1d1c679affac @findepi #14125 ✅ rn ✅ docs
b88d18370f3b784201842748bf68df836d299f72 @findepi #14125 ✅ rn ✅ docs
15dd728c6d2f865bc9ece9672a32df9314b7a4e7 @findepi #14118 ✅ rn ✅ docs
9f0ace6f9312082e8d07d41e78cc8ec15653307f @raunaqmorarka #14137 ✅ rn ✅ docs
cfc12aa6cfc4e5c2b383fcb87aaab0c115e2f65e @martint #14166 ✅ rn ✅ docs

## 17 September 2022 ✅ 

a7e81341b310cc39866153ff657a5d20782e011a @findepi #14074 ✅ rn ✅ docs

## 18 September 2022 ✅

9a04fca2d7182d64f610c0dbf507a0fd23ea7061 @dain #14092 ✅ rn ✅ docs
08342e58c5e44a7b1e7506b339dc300c48cf573e @dain #14092 ✅ rn ✅ docs
761ffbd5657a31744a53f635f6bda6dd25fc6d9b @dain #14092 ✅ rn ✅ docs
2567afef14dc352722d2fbd3382f293c085698b0 @dain #14092 ✅ rn ✅ docs
0a0e62855634ff1b991d45c2fcc6d5d63a98edb4 @dain #14092 ✅ rn ✅ docs
aa225ae4bc78683a9bd65901e4b28d07ff35ece6 @dain #14092 ✅ rn ✅ docs
52eb874c0d4e07ece00225f43274e1652514137b @dain #14092 ✅ rn ✅ docs

## 19 September 2022 ✅ 

e8a1a79b28db17e57d9122d4756b8784952a941e @findepi #14171 ✅ rn ✅ docs
2c92f9f5ce8fb13ead71a02d28ef67df8ee9c5c4 @findepi #14161 ✅ rn ✅ docs
a3d2c2f30e033aca69e6f8d32b460756c1589542 @findepi #14161 ✅ rn ✅ docs
8d70892190639c6677d64eea104586c310adfd9a @findepi #14161 ✅ rn ✅ docs
c59633d0c29d3993553e14ea42e390972ecc0d1c @findepi #14179 ✅ rn ✅ docs
390f4c81e6b554996c553f04d26e37e77c96d6b5 @findepi #14168 ✅ rn ✅ docs
9665ac6beeb0605f1e4e0ea19819c89a08502b5c @findepi #14188 ✅ rn ✅ docs
5dd95a9aae7e75abba95e1fae0a793a0486369b6 @kokosing #14149 ✅ rn ✅ docs
30d642d901baf443bc454617b02864c4b9ab3c0c @martint #13532 ✅ rn ✅ docs
b9b373cfb2fe686b60b582fb5b9080b22213a847 @martint #14182 ✅ rn ✅ docs
30d9b3c70900636cc51d9050956b8efd4ef14194 @findepi #14173 ✅ rn ✅ docs
59bb5fd29945f0b3dcf310bc21cb7f97943eed5f @findepi #14173 ✅ rn ✅ docs
d6a7ba9b88b7c40b1477415838e780f9010acf6b @findepi #14173 ✅ rn ✅ docs
147fe24ecf73feb9845e325a9c8b301028887019 @findepi #14173 ✅ rn ✅ docs
e736d78de877970e421fc9fdfc1a3d8bb7c5fd60 @findepi #14173 ✅ rn ✅ docs
18d875fbe966553d6cfdedf7c3e2fbe5f909cfa2 @findepi #14173 ✅ rn ✅ docs
23c0eaf5790c96740a1cabe0e2f7e1a4b9d46db0 @findepi #14173 ✅ rn ✅ docs
c92c642d874a6f71fd035f15fc4d507e45e07a39 @findepi #14173 ✅ rn ✅ docs
ea9036fe6a51ea662cd93896bd508eb4303bbabe @findepi #14018 ✅ rn ✅ docs
47c1fb1d719ed4d192e358ac4517dd6fe0554642 @findepi #14190 ✅ rn ✅ docs
79dc367f6ff3b2b7e893bb7c91d02d7b33acb744 @findepi #13875 ✅ rn ✅ docs
c373dba78ff308d703bfea14c8dbf7dddd38a6b3 @findepi #13875 ✅ rn ✅ docs
39e2ff22277842ad30fb0cea6cabdf46dc6a37c6 @findepi #14079 ✅ rn ✅ docs
d562faff02a4e0276aa3c39c425bf51809ac6228 @findepi #14079 ✅ rn ✅ docs
6acfd82701687415fef52e2beb4447a828158609 @findepi #14079 ✅ rn ✅ docs
601a3a723ae1b8a249e6aaf89e33f15c12291aca @findepi #13775 ✅ rn ✅ docs
d40b68cee7b6f8bb3771f5d1f228103a826c6afb @findepi #14186 ✅ rn ✅ docs
22dd82d714a2157d7177236f36a00c8ac3aa3ca1 @findepi #14185 ✅ rn ✅ docs

## 20 September 2022

f2b212b6c56b93b07a1b5ba3fb45bbcb64549cea @findepi #14193 ✅ rn ✅ docs
6a4f1ca3b86f592f0b99de6ef3c9affb6303df7e @findepi #12349 ✅ rn ✅ docs
f10a1f9446c39e7ca50524c16d7868123c83004e @findepi #14187 ✅ rn ✅ docs
9485e428ba90edf424f86d700646929578993983 @findepi #14187 ✅ rn ✅ docs
b0b5af1831d3b815790ee60d538253d6da73e120 @findepi #14207 ✅ rn ✅ docs
c16a430b9d3a39ce7669758f5ee47d011e5058f0 @kokosing #14150 ✅ rn ✅ docs
faa308700723f0295c90ee1714d8c84380e956f3 @findepi #14163 ✅ rn ✅ docs
b9b0485bade6f69f1fd3cbe12267e5bdef0abf7f @findepi #14192 ✅ rn ✅ docs
5ac5fdb5f079f532b28128bf0bf9bf158fb8d55f @findepi #14192 ✅ rn ✅ docs
2b77883684e446907848f907da8abb2e33e635c7 @findepi #14077 ✅ rn ✅ docs
5e066e2c9bf42cecf734cd0f223635eb0ee2a89e @Praveen2112 #14206 ✅ rn ✅ docs
e17dc3f4f2713f31c0e74594fa5fe057133055bc @arhimondr #14040 ✅ rn ✅ docs

## 21 September 2022

ee753bf1b21e891e68862be6eca6d8d27e4e28b3 @Praveen2112 #13662 ✅ rn ✅ docs
e9dad2ce84900de4b72dfa80217ecb38bb8121a5 @findepi #14223 ✅ rn ✅ docs
7dfd4f7e23db11749efa7d928ac50d9b2b13f61f @findepi #14223 ✅ rn ✅ docs
8fe8a9114c17b1f4026bb9434607a5c05c44ac03 @findepi #14172 ✅ rn ✅ docs
ccfcb9db2ec7894d9b6c6c796257828471e01c28 @kokosing #13490 ✅ rn ✅ docs
581842e856fbaba01926713792a94c5bb64e8843 @findepi #14198 ✅ rn ✅ docs
1115c56219dad415882f4eca392f5dcf27412d22 @findepi #14234 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-397.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Release 397 (21 Sep 2022)
+
+## General
+
+* Fix incorrect parsing of invalid values in cast from `varchar` to `timestamp`. ({issue}`14164`)
+* Fix potential incorrect results for queries where all data scanned from tables
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-397.md:6]: not sure what "potential" incorrect results is supposed to indicate ...

also dont know if we need to explain leaf stage and downstream more or somehow make this more consumable.

File: docs/src/main/sphinx/release/release-397.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  was processed in the leaf stage and was not passed downstream. ({issue}`14168`)
+* Fix `EXPLAIN (TYPE IO)` query failure for certain queries with redundant table
+  scans. ({issue}`10398`)
+
+## JDBC driver
+
+* Fix potential memory leak when cancelling statements. ({issue}`14176`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-397.md:13]: drop "potential" maybe?

File: docs/src/main/sphinx/release/release-397.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+## Delta Lake connector
+
+* Add `parquet.optimized-writer.enabled` as the new configuration property for
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-397.md:17]: Do we need to add if it defaults to true or false in this context?

File: docs/src/main/sphinx/release/release-397.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  enabling the optimized parquet writer. Deprecate the
+  `parquet.experimental-optimized-writer.enabled` configuration property. ({issue}`14137`)
+
+## Hive connector
+
+* Add `parquet.optimized-writer.enabled` as the new configuration property for
+  enabling the optimized parquet writer. Deprecate the
+  `parquet.experimental-optimized-writer.enabled` configuration property. ({issue}`14137`)
+* Add support for Hive splits for JSON inputs in S3 Select. ({issue}`14040`)
+* Improve planning time for Glue tables with a large number of columns. ({issue}`14206`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-397.md:27]: Glue tables?? You mean tables in S3 or wherever when the metadata is in Glue .. so thats not a Glue table then right?

File: docs/src/main/sphinx/release/release-397.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+* Fix query failure when reading from a Hive view and
+  `hive.hive-views.run-as-invoker` and `hive.hive-views.legacy-translation` are
+  both enabled. ({issue}`14077`)
+
+## Iceberg connector
+
+* Improve performance of queries that contain predicates involving `date_trunc`
+  with an `hour` unit on `date`, `timestamp` or `timestamp with time zone`
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-397.md:35]:
```suggestion
  with an `hour` unit on `date`, `timestamp`, or `timestamp with time zone`
```

File: docs/src/main/sphinx/release/release-397.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  partition columns. ({issue}`14161`)
+* Reduce query latency when using the Glue catalog. ({issue}`13875`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-397.md:37]:
```suggestion
* Reduce query latency when using a Glue catalog. ({issue}`13875`)
```

or 

```suggestion
* Reduce query latency when using Glue catalog. ({issue}`13875`)
```

also could add "for metadata" ??

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-397.md (hunk 3 above)
Comment (raunaqmorarka) [reply docs/src/main/sphinx/release/release-397.md:17]: The defaults are the same as the existing property, only the name has changed

File: docs/src/main/sphinx/release/release-397.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+## Delta Lake connector
+
+* Add `parquet.optimized-writer.enabled` as the new configuration property for
+  enabling the optimized parquet writer. Deprecate the
+  `parquet.experimental-optimized-writer.enabled` configuration property. ({issue}`14137`)
+
+## Hive connector
+
+* Add `parquet.optimized-writer.enabled` as the new configuration property for
```
Comment (raunaqmorarka) [inline docs/src/main/sphinx/release/release-397.md:23]: I forgot to mention here that the catalog session property `experimental_parquet_optimized_writer_enabled` is also replaced by `parquet_optimized_writer_enabled`. The same is true for delta and iceberg as well.

File: docs/src/main/sphinx/release/release-397.md (hunk 2 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-397.md:13]: I think we've used the word "potential" a lot for things that occur sporadically, under specific circumstances, or both. The memory leak doesn't happen every time you cancel a statement, so we need to hedge it.

Same generally applies for the comment you made above.

File: docs/src/main/sphinx/release/release-397.md (hunk 1 above)
Comment (martint) [reply docs/src/main/sphinx/release/release-397.md:6]: We don't need to provide those details. If someone wants to find out more, they can look at the PR. However, the RN entry should explain the conditions under which the correctness issue might happen (type of queries, etc)

File: docs/src/main/sphinx/release/release-397.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
+# Release 397 (21 Sep 2022)
+
+## General
+
+* Fix incorrect parsing of invalid values in cast from `varchar` to `timestamp`. ({issue}`14164`)
+* Fix potential incorrect results for queries where all data scanned from tables
+  was processed in the leaf stage and was not passed downstream. ({issue}`14168`)
+* Fix `EXPLAIN (TYPE IO)` query failure for certain queries with redundant table
```
Comment (martint) [inline docs/src/main/sphinx/release/release-397.md:8]: What's a "redundant table scan" (especially, what does it mean to a user)?

File: docs/src/main/sphinx/release/release-397.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  scans. ({issue}`10398`)
+
+## JDBC driver
+
+* Fix potential memory leak when cancelling statements. ({issue}`14176`)
+
+## Delta Lake connector
+
+* Add `parquet.optimized-writer.enabled` as the new configuration property and
```
Comment (martint) [inline docs/src/main/sphinx/release/release-397.md:17]: How about simply:

> Rename the xxx configuration property to yyy and the xxx session property to yyy.

File: docs/src/main/sphinx/release/release-397.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+  `experimental_parquet_optimized_writer_enabled` session property. ({issue}`14137`)
+
+## Hive connector
+
+* Add `parquet.optimized-writer.enabled` as the new configuration property and
+  `parquet_optimized_writer_enabled` as the new session property for enabling
+  the optimized parquet writer. Deprecate the
+  `parquet.experimental-optimized-writer.enabled` configuration property and
+  `experimental_parquet_optimized_writer_enabled` session property. ({issue}`14137`)
+* Add support for Hive splits for JSON inputs in S3 Select. ({issue}`14040`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-397.md:30]: What does this mean? What's the impact or user-visible effect of this change?

From the description in the issue:

> Trino client will return results faster when S3 Select pushdown is enabled for uncompressed JSON files:
set SESSION hive.s3_select_pushdown_enabled=true;

So I would just say:

> Improve performance when querying JSON data and Hive S3 Select pushdown is enabled.

File: docs/src/main/sphinx/release/release-397.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+* Improve planning time for tables with a large number of columns which use
```
Comment (martint) [inline docs/src/main/sphinx/release/release-397.md:31]: Maybe:

> Improve planning performance when querying tables in the Glue catalog that contain a large number of columns.

File: docs/src/main/sphinx/release/release-397.md (hunk 1 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-397.md:6]: My understanding was that this was the condition in which it happened. Let me try again with info from the PR, may need another suggestion/edit.

File: docs/src/main/sphinx/release/release-397.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+## General
+
+* Fix incorrect parsing of invalid values in cast from `varchar` to `timestamp`. ({issue}`14164`)
+* Fix potential incorrect results for queries with a partitioned output which
+  doesn't depend on any column data. ({issue}`14168`)
+* Fix `EXPLAIN (TYPE IO)` query failure for certain queries on empty tables. ({issue}`10398`)
+
+## Security
+
+* Add support for multiple recipients with JSON web token authentication. ({issue}`13442 `)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-397.md:12]: JWT authentication

File: docs/src/main/sphinx/release/release-397.md [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+* Fix OAuth token refresh causing JSON web token authentication failure. ({issue}`13575`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-397.md:13]: OAuth 2.0 token 


to be consistent with rest of docs

Comment (mosabua) [reply docs/src/main/sphinx/release/release-397.md:13]: and same

JWT authentication failure

File: docs/src/main/sphinx/release/release-397.md [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+
+## JDBC driver
+
+* Fix potential memory leak when cancelling statements. ({issue}`14176`)
+
+## Delta Lake connector
+
+* Rename the `parquet.experimental-optimized-writer.enabled` configuration
+  property and `experimental_parquet_optimized_writer_enabled` session property
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-397.md:22]:
```suggestion
  property and the `experimental_parquet_optimized_writer_enabled` session property
```

File: docs/src/main/sphinx/release/release-397.md [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+  to `parquet.optimized-writer.enabled` and `parquet_optimized_writer_enabled`,
+  respectively. ({issue}`14137`)
+
+## Hive connector
+
+* Rename the `parquet.experimental-optimized-writer.enabled` configuration
+  property and `experimental_parquet_optimized_writer_enabled` session property
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-397.md:29]:
```suggestion
  property and the `experimental_parquet_optimized_writer_enabled` session property
```

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-397.md (hunk 14 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-397.md:22]: This is fine as-is. You don't need to repeat definite articles in lists, e.g. "The cat and the dog are playing" is unnecessarily verbose compared to "the cat and dog are playing."

Comment (mosabua) [reply docs/src/main/sphinx/release/release-397.md:22]: Hmm.. sounds weird to me but oh well

**Review by mosabua — APPROVED** [review]
Looks good now

File: docs/src/main/sphinx/release/release-397.md [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+
+* Improve performance of queries that contain predicates involving `date_trunc`
+  with an `hour` unit on `date`, `timestamp`, or `timestamp with time zone`
+  partition columns. ({issue}`14161`)
+* Improve performance of reads after a `DELETE` removes all rows from a file. ({issue}`14198`)
+* Reduce query latency when using a Glue catalog for metadata. ({issue}`13875`)
+* Fix materalized views temporarily appearing empty when a refresh is about to
+  complete. ({issue}`14145`)
+* Fix potential table corruption when changing a table before committing to the
+  Hive metastore has completed. ({issue}`14118`)
```
Comment (findepi) [inline docs/src/main/sphinx/release/release-397.md:52]: To me, this wording doesn't convey the meaning. 
It sounds like user "changing the table" could trigger the bug, while the triggering condition was e.g. network problems between Trino and HMS.

Any reason why the wording proposed in https://github.com/trinodb/trino/issues/14154#issuecomment-1249814960 was not used here?
I checked PR comments of this PR (https://github.com/trinodb/trino/pull/14194), but didn't spot any related discussion that would explain this.
